core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -1053,13 +1053,17 @@ private int compareHighBound(RangeView that)
                 return Optional.of(this);
             }
 
+            boolean merge;
             if (next.isLowUnbounded()) {
-                return Optional.of(next);
+                // both are low-unbounded
```

Comment:
why? no check that `this` is unbouded

Reply:
Otherwais it check above would fail

Reply:
Which check? `if (this.isHighUnbounded()) {` or `if (this.compareTo(next) > 0) {`?

Reply:
`this.compareTo(next)`

this cannot be greater than next. So if `next` is `isLowUnbounded` (the lowest possible) so must be this.

Reply:
this could be `[1, 2]` and next could be `(-inf, 3)`, correct?

Reply:
this is checked on the first line of the method